src/fileimpl.cpp

```diff
@@ -294,14 +294,12 @@ class Grouping
     //    in the call stack.
     // 2. With `glibc` an exceptional execution of `std::call_once` doesn't
     //    unlock the mutex associated with the `std::once_flag` object.
+    std::lock_guard<std::mutex> lock(m_direntLookupCreationMutex);
     if ( !m_direntLookup ) {
-      std::lock_guard<std::mutex> lock(m_direntLookupCreationMutex);
-      if ( !m_direntLookup ) {
-        if (m_direntLookupSize == 0) {
-          m_direntLookup = std::make_unique<DirentLookup>(mp_pathDirentAccessor.get());
-        } else {
-          m_direntLookup = std::make_unique<FastDirentLookup>(mp_pathDirentAccessor.get(), m_direntLookupSize);
-        }
+      if (m_direntLookupSize == 0) {
+        m_direntLookup = std::make_unique<DirentLookup>(mp_pathDirentAccessor.get());
+      } else {
+        m_direntLookup = std::make_unique<FastDirentLookup>(mp_pathDirentAccessor.get(), m_direntLookupSize);
       }
```

Comment:
Can't you use an atomic variable `m_direntLookupIsInitialized` to avoid locking on each access?